spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDeploymentController.java

```diff
@@ -98,10 +98,8 @@ public StreamDeploymentController(StreamDefinitionRepository repository, StreamS
 	 */
 	@RequestMapping(value = "/{name}", method = RequestMethod.DELETE)
 	public ResponseEntity<Void> undeploy(@PathVariable("name") String name) {
-		StreamDefinition stream = this.repository.findOne(name);
-		if (stream == null) {
-			throw new NoSuchStreamDefinitionException(name);
-		}
+		this.repository.findById(name)
```

Comment:
We could also add a custom repository method as we did in Skipper, such as `findByIdRequired`

Reply:
I don't think that's a good idea as I'm not really fan of adding custom methods which just throws something.